scripts/restart-server

```diff
@@ -78,24 +78,25 @@ else:
     logging.info("%s Tornado process", verbing)
     subprocess.check_call(["supervisorctl", action, "zulip-tornado", "zulip-tornado:*"])
 
-# Restart the uWSGI and related processes via supervisorctl.
+# Restart the uWSGI and related processes via supervisorctl.  This
+# results in a different stop ordering than `stop-server`, because
+# this is concerned with minimizing downtime, not with quiescing the
+# server as completely as possible.
+services = ["zulip-workers:*", "zulip-django"]
 if action == "restart":
-    logging.info("Stopping workers")
-    subprocess.check_call(["supervisorctl", "stop", "zulip-workers:*"])
-    logging.info("Stopping server core")
-    subprocess.check_call(["supervisorctl", "stop", *core_server_services])
+    logging.info("Stopping server")
+    subprocess.check_call(["supervisorctl", "stop", *services])
```

Comment:
How have you tested this change?  My past experience with supervisord has been that `supervisorctl stop` with large blocks of processes in the list is slower than doing multiple smaller runs, which is why we have them split right now.  However, it's possible that they've fixed their bug.  (I remember this being reproducible with complete dummy jobs that just sleep forever, but we probably want to test with `staging` or something, `time supervisorctl stop a b` vs `time supervisorctl stop a; supervisorctl stop b`).

b2d01e2da0a90e3de8a9919e0261da1c261c68c3 appears to be the background, though I remember rechecking this assumption in 2016/2017 or so and it still being sadly true.

Reply:
See https://github.com/zulip/zulip/issues/14959#issuecomment-823560123